webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,297 @@
+import React, {ReactNode} from 'react';
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+import {TicketData, TicketDetails} from 'types/tooltip';
+import DefaultAvatar from 'components/default_avatar/default_avatar';
+
+import './ticketStyle.scss';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    ticketDetails?: TicketDetails | null;
+    connectedInstances: Instance[];
+    fetchIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+    }>;
+}
+
+export type State = {
+    ticketId: string;
+    ticketDetails?: TicketDetails | null;
+};
+
+const isAssignedLabel = ' is assigned';
+const unAssignedLabel = 'Unassigned';
+const jiraTicketSummaryMaxLength = 80;
+const maxTicketDescriptionLength = 160;
+
+enum myStatus {
+    INDETERMINATE = 'indeterminate',
+    DONE = 'done',
+}
+
+const myStatusClasses: Record<string, string> = {
+    [myStatus.INDETERMINATE]: 'ticket-status--indeterminate',
+    [myStatus.DONE]: 'ticket-status--done',
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        const issueKey = this.getIssueKey();
+        let ticketID = '';
+        if (issueKey) {
+            ticketID = issueKey.ticketID;
+        }
+
+        this.state = {
+            ticketId: ticketID,
+        };
+    }
+
+    getIssueKey = () => {
+        let ticketID = '';
+        let instanceID = '';
+
+        for (const instance of this.props.connectedInstances) {
+            instanceID = instance.instance_id;
+
+            if (!this.props.href.includes(instanceID)) {
+                continue;
+            }
+
+            // We already check href.includes above in the if statement before this try block
+            try {
+                const regex = /(https|http):\/\/.*\/.*\?.*selectedIssue=([\w-]+)&?.*|(https|http):\/\/.*\/browse\/([\w-]+)?.*/;
+                const result = regex.exec(this.props.href);
+                if (result) {
+                    ticketID = result[2] || result[4];
+                    return {ticketID, instanceID};
+                }
+                break;
+            } catch (e) {
+                break;
+            }
+        }
+
+        return null;
+    }
+
+    isUserConnectedAndStateNotLoaded() {
+        const {connected} = this.props;
+        const {ticketDetails} = this.state;
+
+        return connected && ticketDetails;
```

Comment:
Since this method name implies that we are returning a boolean

```suggestion
        return Boolean(connected && ticketDetails);
```